### ISSUE TITLE
`ariaHideOutside`: Modal with flat combobox (no popover) put the page in aria-hidden

### ISSUE THREAD

**Issue by Eldraed** [adobe/react-spectrum#8934]
### Provide a general summary of the issue here

In our DS we have duplicated something that you made on react-spectrum as an alpha: SearchAutocomplete.
Your component when on desktop is opening as a Popover, but when on mobiles, it opens a modal with a flat combobox in it, which means no popover (useCombobox is used for both desktop/mobile).

Since your latest changes on the `ariaHideOutside` to use `inert` and `aria-hidden` depending on the cases (modal/popover => inert, combobox => aria-hidden), it creates an issue in the mobile version of the SearchAutocomplete, after closing the modal, `aria-hidden` stays.

### 🤔 Expected Behavior?

`aria-hidden` should be removed after closing the modal

### 😯 Current Behavior

`aria-hidden` stays when closing the modal

### 💁 Possible Solution

It seems from what I understood that:
- `ariaHideOutside` is triggered first from the `useCombobox`
  - it adds `aria-hidden` to all external elements other than the combobox
- `ariaHideOutside` is then triggered from the Modal
  - it adds `inert` to all external elements of the modal

When closing:
- the callback on closing of the `ariaHideOutside` from the combobox is having `count` > 1 here:
https://github.com/adobe/react-spectrum/blob/eba6cc03fc30e291db60cf88d3bc0429a75c2f4d/packages/%40react-aria/overlays/src/ariaHideOutside.ts#L188
  - it ignores the `aria-hidden` set by the combobox, thinking that the next callback from the modal will take care of it
- the callback from the `ariaHideOutside` of the modal when closing is running well, but since it sets `inert` and not `aria-hidden`, it only removes `inert` attributes
- `aria-hidden` from the `useCombobox` stays

### 🔦 Context

In my opinion, it's both a bug and a bad implementation.

- Maybe there are other cases where a second `ariaHideOutside` could be called inside of a modal with `aria-hidden` other than this specific use case.
- Using `useCombobox` as flat component (without a Popover) is a bad implementation
  - We should switch to `useAutocomplete`, but it means switching from `useComboboxState` to `useAutocompleteState` depending on if we are on desktop/mobile, which is in my opinion not a good pattern too, having one state for both would be better, but I don't think it's possible due to the differences in the two APIs.

Our react-aria versions are the one when `inert` appeared in `ariaHideOutside`, I saw some changes on the code due to this issue: https://github.com/adobe/react-spectrum/issues/8730
But the fix only solve the reverse case: https://github.com/adobe/react-spectrum/pull/8739/files
We should remove the `aria-hidden` when removing the `inert` here:
https://github.com/adobe/react-spectrum/blob/eba6cc03fc30e291db60cf88d3bc0429a75c2f4d/packages/%40react-aria/overlays/src/ariaHideOutside.ts#L54

**The only workaround I have** currently is to trigger `ariaHideOutside` on the same scope as the `useCombobox` before it, with `{ shouldUseInert: true }`, because it will consider it as already hidden.

### 🖥️ Steps to Reproduce

You can reproduce on your own documentation/storybook (non public documentation as your component is in `alpha` since years for Spectrum 😬

- Open this page (one of the latest links generated in one of your PRs): https://reactspectrum.blob.core.windows.net/reactspectrum/abe3409cd0edeb4b4cd64df04e1cc8ac9631e31a/docs/react-spectrum/SearchAutocomplete.html
(Can be also check on the [Storybook](https://reactspectrum.blob.core.windows.net/reactspectrum/abe3409cd0edeb4b4cd64df04e1cc8ac9631e31a/storybook/iframe.html?scrolling=false&providerSwitcher-locale=&providerSwitcher-theme=&providerSwitcher-scale=&providerSwitcher-express=false&globals=&args=&id=searchautocomplete--default&viewMode=story))
- Put you in mobile mode from the devtools
- The page is not in `aria-hidden`
- Open the autocomplete
- Close the autocomplete
- The page stays in `aria-hidden`

You need to be on a fresh page to see the behavior as the `aria-hidden` stays after.

https://github.com/user-attachments/assets/ec4ee202-db41-4a66-a893-49275033c400

Some debugging:

https://github.com/user-attachments/assets/40d8a0f4-9eb6-490c-8fa3-97930c548fad

When there is `input` in `visibleNodes`, it's the `useComboBox` callback from `ariaHideOutside`, otherwise it's the modal one.
It doesn't remove the `aria-hidden` on nodes that are duplicates of the modal `ariaHideOutside` from what I understand. So only nodes that are between the modal and the combobox elements are having the `aria-hidden` removal

### Version

"@react-aria/overlays": "3.28.0", "@react-aria/combobox": "3.13.0"

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

MacOS Tahoe

**Comment by snowystinger** [comment]
Apologies, SearchAutocomplete shouldn't be used as a basis to build anything. It's an early alpha that was pretty much abandoned. That's why you can't find it in the published documentation. There are some left over files, but they should really be removed.

Instead, you should use the new and documented `Autocomplete` component, it's in Beta but should move to RC very soon https://react-spectrum.adobe.com/react-aria/Autocomplete.html#autocomplete

We do aim to expand the Autocomplete's functionality to support combobox as well in which case we'll probably rework our current combobox implementation to be based on Autocomplete. I don't have a timeline for that right now though. So in the meantime, you could render a ComboBox for desktop, but an Autocomplete for mobile.

**Comment by Eldraed** [comment]
Hello @snowystinger, thanks for the feedback.

Sorry I forgot to add a bit more information, we have done this "duplication" of SearchAutocomplete 2 years ago when `useAutocomplete` wasn't even there 😬 We found the alpha version on Spectrum while searching in react-aria code an example for Autocomplete, but we were aware that it would had an unsure future 😅
That's why we are in this situation today, we would have done things differently otherwise with now `useAutocomplete`.

Well, that's what I was thinking of doing (using `useAutocomplete`) depending on your answer, but as the states hooks have different APIs (for autocomplete and combobox) it might need some workarounds too to be able to switch safely between both on resize until you rework eventually the `useCombobox` to be based on `useAutocomplete`.

**Just to be sure:** So, are we sure that no other cases would benefit of solving this behavior in `ariaHideOutside` ? `ariaHideOutside` being called inside of a Modal without the `shouldUseInert` ?
React-aria side it doesn't seems to, I don't know if some people use `ariaHideOutside` on their own ?

If you want to stay like that, then we can close this issue and say that you won't solve the `ariaHideOutside` behavior in that case 🙂

Edit, I didn't search for this sorry but: Is there a specific reason of not using `shouldUseInert` on the `useCombobox` ?

**Comment by snowystinger** [comment]
> So, are we sure that no other cases would benefit of solving this behavior in ariaHideOutside ?

A good question, I don't currently have a use case. I'm inclined to close this until such a time.

> Is there a specific reason of not using shouldUseInert on the useCombobox

We don't use it because then you wouldn't be able to tab to the next element for instance. You'd just be stuck in the input.

**Comment by Eldraed** [comment]
Ok, let's close this in that case !

Thanks for the info regarding tabbing. I'll keep my workaround that adds `inert`, as it's working fine in my case and tabbing is still working as expected (maybe an edge case). But I'm planning switching out of the flat combobox by using `useAutocomplete` instead.

Thanks for your help 🙂
